cachi2/core/models/input.py

```diff
@@ -172,11 +117,52 @@ def _requirements_file_path_is_relative(cls, paths: list[Path]) -> list[Path]:
         return paths
 
 
+class _ExtraOptions(pydantic.BaseModel, extra="forbid"):
```

Comment:
Just to clarify: is this model intended to be used only within `RpmPackageInput`, or is it the intention to have it as possibility for other package managers to adopt if needed?

If I think of the latter, I imagine one could use `{type: pip, options: {dnf: {}}}`, and it would be valid, even though it's a noop.

Reply:
> If I think of the latter, I imagine one could use {type: pip, options: {dnf: {}}}, and it would be valid, even though it's a noop.

True. Although this is IMO not the time to deal with that, because we're not enabling it globally. Once we decide to do so, we could e.g. do:

```
class _PackageInputBase(pydantic.BaseModel, extra="forbid"):
     """Common input attributes accepted for all package types."""
 
     type: PackageManagerType
     path: Path = Path(".")
+    options: Optional[_ExtraOptions] = None
+    _option_namespaces: list[str] = []
 
     @pydantic.field_validator("path")
     def _path_is_relative(cls, path: Path) -> Path:
         return check_sane_relpath(path)
 
+    @pydantic.model_validator(mode="after")
+    def _validate_pkgmgr_type_option_pairings(self) -> Any:
+        breakpoint()
+        if self.options is None:
+            return self
+
+        if (diff := self.options.model_fields_set - set(self._option_namespaces)):
+            raise ValueError(
+                f"Unsupported options for type:'{self.type}': {diff} | supported options: '{self._option_namespaces}'")
+        return self
+
```

AND
```
 class RpmPackageInput(_PackageInputBase):
     """Accepted input for a rpm package."""
 
     type: Literal["rpm"]
     options: Optional[_ExtraOptions] = None
+    _option_namespaces: list[str] = ["dnf"]
```

and deal with this on the global validator level. For the record I tried the above with your example and got:
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for tagged-union[function-after[_validate_pkgmgr_type_option_pairings(), BundlerPackageInput],function-after[_validate_pkgmgr_type_option_pairings(), GomodPackageInput],function-after[_validate_pkgmgr_type_option_pairings(), NpmPackageInput],function-after[_validate_pkgmgr_type_option_pairings(), PipPackageInput],function-after[_validate_pkgmgr_type_option_pairings(), RpmPackageInput],function-after[_validate_pkgmgr_type_option_pairings(), YarnPackageInput]]
E       pip
E         Value error, Unsupported options for type:'pip': {'dnf'} | supported options: '[]' [type=value_error, input_value={'type': 'pip', 'requirem...': {'dnf': {'foo': {}}}}, input_type=dict]
```

Reply:
Fair enough. So maybe add a small docstring stating we intend this to be a generic model for all package managers?